### ISSUE TITLE
SPGEMM color algorithms not implemented, _spgemm_impl_color.hpp not used

### ISSUE THREAD

**Issue by brian-kelley** [kokkos/kokkos-kernels#677]
The file ``KokkosSparse_spgemm_impl_color.hpp`` is included in ``KokkosSparse_spgemm_impl.hpp``, but nothing defined in it is used anywhere. The SPGEMM handle actually defines algorithm enum values SPGEMM_KK_COLOR, SPGEMM_KK_MULTICOLOR and SPGEMM_KK_MULTICOLOR2, but those enum constants don't appear anywhere else (I don't know what would happen if you tried to use them, probably just a crash or an empty product).

@srajama1 Is this something we can get rid of? Do you know what Mehmet's idea was in using D2 coloring in SPGEMM? I could try to understand the code if it's of interest.

**Comment by srajama1** [comment]
I think this was added to try out work by McCourt et al. which uses d-2 coloring on structure C for SpGEMM. I don't remember the exact details if we actually tried it and not chose to move to production or if we never tried it. It is quite possible it is the later as we didn't have a good d-2 coloring on GPUs at that point anyway. I think we can remove the dummy arguments now. We can add it back, if we decide to try it again.
